src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/CollectionCodecs.java

```diff
@@ -143,4 +211,22 @@ static JsonNode toJsonNode(
     }
     return result;
   }
+
+  /** Method that will convert from driver-provided CQL Map type into JSON output. */
+  static JsonNode toJsonNodeFromMap(
+      JSONCodec<?, ?> elementCodec0, ObjectMapper objectMapper, Object mapValue)
+      throws ToJSONCodecException {
+    JSONCodec<?, Object> elementCodec = (JSONCodec<?, Object>) elementCodec0;
+    final ObjectNode result = objectMapper.createObjectNode();
+    for (Map.Entry<Object, Object> entry : ((Map<Object, Object>) mapValue).entrySet()) {
+      final String key = String.valueOf(entry.getKey());
```

Comment:
Are we validating anywhere to not read if map key is not text? Not all cassandra data type can be supported with `String.valueOf(entry.getKey())`

Reply:
Added checking that we indeed get `String` key, throw exception if not.